lib/spack/spack/build_systems/autotools.py

```diff
@@ -248,39 +248,143 @@ def _set_autotools_environment_variables(self):
         """
         os.environ["FORCE_UNSAFE_CONFIGURE"] = "1"
 
+    @run_before('configure')
+    def _do_patch_libtool_configure(self):
+        """Patch bugs that propagate from libtool macros into "configure" and
+        further into "libtool". Note that patches that can be fixed by patching
+        "libtool" directly should be implemented in the _do_patch_libtool method
+        below."""
+
+        # Exit early if we are required not to patch libtool-related problems:
+        if not self.patch_libtool:
+            return
+
+        x = fs.FileFilter(
+            *fs.find(self.build_directory, 'configure', recursive=True))
+
+        # Fix parsing of compiler output when collecting predeps and postdeps
+        # https://lists.gnu.org/archive/html/bug-libtool/2016-03/msg00003.html
+        x.filter(r'^(\s*if test x-L = )("\$p" \|\|\s*)$', r'\1x\2')
+        x.filter(r'^(\s*test x-R = )("\$p")(; then\s*)$',
+                 r'\1x\2 || test x-l = x"$p"\3')
+        # Support Libtool 2.4.2 and older:
+        x.filter(r'^(\s*test \$p = "-R")(; then\s*)$',
+                 r'\1 || test x-l = x"$p"\2')
```

Comment:
For easier reading can we add keywords here:
```suggestion
        x.filter(regex=r'^(\s*if test x-L = )("\$p" \|\|\s*)$', repl=r'\1x\2')
        x.filter(regex=r'^(\s*test x-R = )("\$p")(; then\s*)$',
                 repl=r'\1x\2 || test x-l = x"$p"\3')
        # Support Libtool 2.4.2 and older:
        x.filter(regex=r'^(\s*test \$p = "-R")(; then\s*)$',
                 repl=r'\1 || test x-l = x"$p"\2')
```
and below?